ui/components/src/model/units/bytes.ts

```diff
@@ -27,48 +29,45 @@ export const BYTES_GROUP_CONFIG: UnitGroupConfig = {
   abbreviate: true,
 };
 export const BYTES_UNIT_CONFIG: Readonly<Record<BytesUnitKind, UnitConfig>> = {
+  // This uses units that are powers of 1000.
+  // In other words, 1KB = 1000 bytes.
   Bytes: {
     group: 'Bytes',
     label: 'Bytes',
   },
 };
 
-/**
- * Format value as bytes, use abbreviate option for more readable sizes (KB, MB, GB, etc.)
- * https://stackoverflow.com/questions/15900485/correct-way-to-convert-size-in-bytes-to-kb-mb-gb-in-javascript/18650828#18650828
- */
-export function formatBytes(bytes: number, unitOptions: BytesUnitOptions) {
+export function formatBytes(bytes: number, { abbreviate, decimal_places }: BytesUnitOptions) {
   if (bytes === 0) return '0 bytes';
 
-  // default to full 'Bytes' formatting
-  const options = unitOptions.abbreviate === undefined ? { ...unitOptions, abbreviate: false } : unitOptions;
-
-  let decimalPlaces = options.decimal_places ?? DEFAULT_DECIMAL_PLACES;
-  // avoids minimumFractionDigits value is out of range error, possible values are 0 to 20
+  decimal_places = decimal_places ?? DEFAULT_DECIMAL_PLACES;
+  // Avoids maximumFractionDigits value is out of range error. Possible values are 0 to 20.
   // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#minimumfractiondigits
-  if (decimalPlaces < 0) {
-    decimalPlaces = 0;
-  } else if (decimalPlaces > 20) {
-    decimalPlaces = 20;
+  if (decimal_places < 0) {
+    decimal_places = 0;
+  } else if (decimal_places > 20) {
+    decimal_places = 20;
   }
 
-  const formatParams: Intl.NumberFormatOptions = {
-    style: 'decimal',
-    maximumFractionDigits: decimalPlaces,
-    useGrouping: true,
-  };
-  const formatter = new Intl.NumberFormat('en-US', formatParams);
+  const showFullNumber = abbreviate == false || bytes < 1000;
 
-  if (options.abbreviate === false) {
-    return `${formatter.format(bytes)} bytes`;
+  if (showFullNumber) {
```

Comment:
Out of curiosity, why use `numbro` in one case and `Intl.NumberFormat` in the other? Do we have a rule of thumb where we'd want one over the other or can we standardize on one for most cases?

Reply:
Related: anything worth adding to the newly updated `ui-guidelines` on this if we have some guidelines? At a minimum, it might be worth adding an entry for number formatting to the [list of common libs we use](https://github.com/perses/perses/blob/main/ui/ui-guidelines.md#notable-tooling-choices).

Reply:
Oo good call, will do that

Yeah, there is a limitation in terms of what each can do. `numbro` handles formatting bytes as KB, MB, GB, etc. nicely. As far as I could tell, `numbro` cannot do something like "don't use MB, GB, etc., just write out the full number."

`Intl.NumberFormat`, on the other hand, can easily write out the full number. As far as I can tell, it does not have a way to format bytes as KB, MB, GB, etc. nicely. It has an option called `notation: compact`, but this will use units that correspond to millions and billions, so we would end up with KB, MB, BB 😄

Overall, it's looking like `numbro` does the formatting that we really need with a pretty readable API, and so I'm sensing that ideally we'd move toward consolidation on `numbro`. I want to test this with the decimal formatting and time formatting to confirm, but that's my sense so far. 

Also, based on discussion today, I'm seeing a possible future where we don't need to worry about having the "full number" in the y-axis. In that possible future, we can remove `Intl.NumberFormat` ✂️

Reply:
Ah gotcha, that makes sense!